pkg/provisioner/ironic/ironic.go

```diff
@@ -1637,27 +1637,33 @@ func (p *ironicProvisioner) IsReady() (result bool, err error) {
 	return checker.IsReady()
 }
 
-func (p *ironicProvisioner) IsProvisioning() bool {
-	_, ok := p.getCurrentProvisioningHosts()[p.host.Name]
-	return ok
-}
+func (p *ironicProvisioner) HasProvisioningCapacity() (result bool, err error) {
+	hosts, err := p.getCurrentProvisioningHosts()
+	if err != nil {
+		return false, err
+	}
 
-func (p *ironicProvisioner) CanBeProvisioned() bool {
-	return len(p.getCurrentProvisioningHosts()) < maxProvisioningHosts
+	// If the current host is already under processing then let's skip the test
+	if _, ok := hosts[p.host.Name]; ok {
+		return true, nil
+	}
+
+	return len(hosts) < maxProvisioningHosts, nil
 }
 
-func (p *ironicProvisioner) getCurrentProvisioningHosts() map[string]struct{} {
+func (p *ironicProvisioner) getCurrentProvisioningHosts() (map[string]struct{}, error) {
 	//lazy init
 	if p.provisioningHosts != nil {
-		return p.provisioningHosts
+		return p.provisioningHosts, nil
 	}
```

Comment:
We don't need this cache any more, do we?

Reply:
Right, removing it